commander/efd.py

```diff
@@ -90,16 +91,84 @@ async def query_efd_timeseries(request):
         response_data = dict(zip(sources, results))
         return web.json_response(response_data)
 
+    async def query_efd_logs(request):
+        global efd_clients
+        req = await request.json()
+
+        try:
+            efd_instance = req["efd_instance"]
+            start_date = req["start_date"]
+            end_date = req["end_date"]
+            cscs = req["cscs"]
+        except Exception:
+            return web.json_response(
+                {"ack": "Some of the required parameters is not present"}, status=400
+            )
+
+        efd_client = efd_clients.get(efd_instance)
+        if efd_client is None:
+            efd_client = connect_to_efd_intance(efd_instance)
+        if efd_client is None:
```

Comment:
Is it correct that this "if" is repeated and with two different results? In the first "if" you assign a value to "efd_client" and in the second you return a function. Could it be that the first if is "if efd_client is not None"?

Reply:
Yes, this is correct 😉️. This works as follows:
1. Check if we have a connected instance saved at `efd_clients`
2. If there is no connected instance (`efd_client is None == True`), then try to connect to the instance
3. If on the previous step the instance could not be connected, then we return an error response with status 400

Reply:
Instances will be connected on demand, this is why we use two different if statements. We could nest the second if on the first one, but in this way it also works